autorest/utility_test.go

```diff
@@ -239,6 +240,36 @@ func TestEncodeWithValidNotPathQuery(t *testing.T) {
 	}
 }
 
+func TestMapToValues(t *testing.T) {
+	m := map[string]interface{}{
+		"a": "a",
+		"b": 2,
+	}
+	v := url.Values{
+		"a": []string{"a"},
+		"b": []string{"2"},
+	}
+	if !reflect.DeepEqual(MapToValues(m), v) {
+		t.Fatalf("autorest: MapToValues method failed to return correct values - expected(%v) got(%v)", v, MapToValues(m))
+	}
+}
+
+func TestMapToValuesWithArrayValues(t *testing.T) {
```

Comment:
indeed this test might be passing because you're initializing `v` with a map literal yourself, try using url.Values.Add, it might fail.

so I think you should test this with `url.Values.Get` (same for `TestMapToValues` above) with each key.